src/opera/pge/base_pge.py

```diff
@@ -238,24 +241,62 @@ def _run_sas_qa_executable(self):
             self.logger.info(self.name, ErrorCode.QA_SAS_PROGRAM_DISABLED,
                              'SAS QA is disabled, skipping')
 
+    def _checksum_output_products(self):
+        """
+        Generates a dictionary mapping output product file names to the
+        corresponding MD5 checksum digest of the file's contents.
+
+        The output products to generated checksums for is determined by scanning
```

Comment:
Maybe rewrite "The output products to generated checksums for"  - I get it, but I had to read it a couple of times.